jsapp/js/attachments/AttachmentActionsDropdown.tsx

```diff
@@ -0,0 +1,143 @@
+      await removeAttachmentMutation.mutateAsync(String(attachment!.id))
+      setIsDeleteModalOpen(false)
+      notify(t('##Attachment_type## deleted').replace('##Attachment_type##', attachmentTypeName))
+      props.onDeleted()
+    } catch (e) {
+      notify(t('An error occurred while removing the attachment'), 'error')
+    } finally {
+      setIsDeletePending(false)
+    }
+  }
+
+  function requestDownloadFile() {
+    // We use `!` to assert that the attachment exists, because this code would be unreachable without it.
+    downloadUrl(attachment!.download_url)
+  }
+
+  // We find the question that the attachment belongs to, to determine the text to display in the modal.
+  const questionType = props.asset.content?.survey?.find((row) => row.$xpath === attachment.question_xpath)?.type
+  let attachmentTypeName = t('attachment')
+  if (questionType === QuestionTypeName.audio) {
+    attachmentTypeName = t('audio recording')
+  } else if (questionType === QuestionTypeName.video) {
+    attachmentTypeName = t('video recording')
+  } else if (questionType === QuestionTypeName.image) {
+    attachmentTypeName = t('image')
+  } else if (questionType === QuestionTypeName['background-audio']) {
+    attachmentTypeName = t('background audio recording')
+  }
+
+  const userCanChangeSubmission = userHasPermForSubmission('change_submissions', props.asset, props.submissionData)
+
+  return (
+    <>
+      {/* We don't use portal here, as opening this inside SubmissionModal causes the menu to open in weird place */}
+      <Menu withinPortal={false} closeOnClickOutside closeOnItemClick position='bottom-end'>
+        <Menu.Target>
+          <span style={{ position: 'relative' }}>
+            <ActionIcon size='md' variant='transparent' iconName='more' />
+          </span>
+        </Menu.Target>
+
+        <Menu.Dropdown>
+          <Menu.Item onClick={requestDownloadFile}>
+            <Icon name='download' /> {t('Download')}
+          </Menu.Item>
+          <Menu.Item variant='danger' onClick={() => setIsDeleteModalOpen(true)} disabled={!userCanChangeSubmission}>
+            <Icon name='trash' /> {t('Delete')}
+          </Menu.Item>
+        </Menu.Dropdown>
+      </Menu>
+
+      <Modal
+        opened={isDeleteModalOpen}
+        onClose={() => {
+          setIsDeleteModalOpen(false)
+        }}
+        title={t('Delete ##attachment_type##').replace('##attachment_type##', attachmentTypeName)}
+      >
+        {/* We don't want "x" button to get focus (see https://mantine.dev/core/modal/#initial-focus) */}
+        <FocusTrap.InitialFocus />
+
+        <Stack>
+          <p>
+            {t('Are you sure you want to delete this ##attachment_type##?').replace(
+              '##attachment_type##',
+              attachmentTypeName,
+            )}
+          </p>
+
+          <Group justify='flex-end'>
+            <Button variant='light' size='lg' onClick={() => setIsDeleteModalOpen(false)} disabled={isDeletePending}>
```

Comment:
It looks like [they specified the delete to be invisible](https://www.notion.so/kobotoolbox/Removing-attachments-8e2eb538680e427fa14173ec523c58e2?pvs=4#592a78e7a68344228b216dc028db6ce8) if the user does not have enough permissions (not disabled)

Reply:
Fixed :)